modules/repository-s3/src/internalClusterTest/java/org/elasticsearch/repositories/s3/S3RepositoryThirdPartyTests.java

```diff
@@ -239,9 +240,26 @@ public void testReadFromPositionLargerThanBlobLength() {
 
         long position = randomLongBetween(blobBytes.length(), Long.MAX_VALUE - 1L);
         long length = randomLongBetween(1L, Long.MAX_VALUE - position);
-        var exception = expectThrows(AmazonClientException.class, () -> readBlob(repository, blobName, position, length));
 
-        assertThat(exception, instanceOf(AmazonS3Exception.class));
-        assertThat(((AmazonS3Exception) exception).getStatusCode(), equalTo(RestStatus.REQUESTED_RANGE_NOT_SATISFIED.getStatus()));
+        var exception = expectThrows(UncategorizedExecutionException.class, () -> readBlob(repository, blobName, position, length));
+        assertThat(exception.getCause(), instanceOf(ExecutionException.class));
+        assertThat(exception.getCause().getCause(), instanceOf(IOException.class));
+        assertThat(
+            exception.getCause().getCause().getMessage(),
+            containsString(
+                "Requested range [start="
+                    + position
+                    + ", end="
+                    + (position + length - 1L)
+                    + ", currentOffset=0] cannot be satisfied for blob object ["
+                    + repository.basePath().buildAsString()
+                    + blobName
+                    + ']'
+            )
+        );
+
+        var cause = exception.getRootCause();
+        assertThat(cause, instanceOf(AmazonS3Exception.class));
+        assertThat(((AmazonS3Exception) cause).getStatusCode(), equalTo(RestStatus.REQUESTED_RANGE_NOT_SATISFIED.getStatus()));
```

Comment:
nit/one-liner suggestion:

```suggestion
        assertThat(
            asInstanceOf(AmazonS3Exception.class, exception.getRootCause()).getStatusCode(),
            equalTo(RestStatus.REQUESTED_RANGE_NOT_SATISFIED.getStatus())
        );
```

Reply:
TIL, I pushed 42ab562ca6097027876dbf8d90839d97d3d2567e